test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -362,92 +364,79 @@ public final void testUnknownObjectException() throws IOException {
+            logger.trace("test query: {}", testQuery);
+
+            try {
+                parseQuery(testQuery);
+                if (expectedException) {
+                    fail("some parsing exception expected for query: " + testQuery);
                 }
-                //inject the start of the new object
-                String testQuery = validQuery.substring(0, insertionPosition) + "{ \"newField\" : ";
-                String secondPart = validQuery.substring(insertionPosition);
-                int currentLevel = level;
-                boolean quotes = false;
-                for (int i = 0; i < secondPart.length(); i++) {
-                    if (secondPart.charAt(i) == '"') {
-                        quotes = quotes == false;
-                    } else if (quotes == false && secondPart.charAt(i) == '{') {
-                        currentLevel++;
-                    } else if (quotes == false && secondPart.charAt(i) == '}') {
-                        currentLevel--;
-                        if (currentLevel == level) {
-                            //close the additional object in the right place
-                            testQuery += secondPart.substring(0, i - 1) + "}" + secondPart.substring(i);
-                            break;
-                        }
-                    }
+            } catch (ParsingException | ElasticsearchParseException e) {
+                // different kinds of exception wordings depending on location
+                // of mutation, so no simple asserts possible here
+                if (expectedException == false) {
+                    throw new AssertionError("unexpected exception when parsing query:\n" + testQuery, e);
                 }
-
-                try {
-                    parseQuery(testQuery);
-                    if (expectedException) {
-                        fail("some parsing exception expected for query: " + testQuery);
-                    }
-                } catch (ParsingException | ElasticsearchParseException e) {
-                    // different kinds of exception wordings depending on location
-                    // of mutation, so no simple asserts possible here
-                    if (expectedException == false) {
-                        throw new AssertionError("unexpected exception when parsing query:\n" + testQuery, e);
-                    }
-                } catch(IllegalArgumentException e) {
-                    assertThat(e.getMessage(), containsString("unknown field [newField], parser not found"));
-                    if (expectedException == false) {
-                        throw new AssertionError("unexpected exception when parsing query:\n" + testQuery, e);
-                    }
+            } catch (IllegalArgumentException e) {
+                assertThat(e.getMessage(), containsString("unknown field [newField], parser not found"));
+                if (expectedException == false) {
```

Comment:
I think the `assertThat` statement above this `if` statement should actually be below it?